manager-bundle/src/Process/ProcessFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerBundle\Process;
+
+use Symfony\Component\Process\Process;
+
+/**
+ * @internal
+ */
+class ProcessFactory
```

Comment:
We need this class to make the command better testable, right?

Reply:
Exactly. Open for better ideas of course 🙂

Reply:
How about using an optional closure argument to create the process?

Reply:
What do you mean by this?

Reply:
I think @aschempp means to adjust the `InitializeApplicationCommand` like this:

```php
public function __construct(string $projectDir, string $webDir, Filesystem $filesystem = null, callable $processFactory = null)
{
    $this->processFactory = $processFactory;

    // …
}

private function executeCommand(array $command, OutputInterface $output): void
{
    if ($this->processFactory) {
        $process = $this->processFactory->create(array_merge([$this->phpPath, $this->consolePath], $command));
    } else {
        $process = new Process(array_merge([$this->phpPath, $this->consolePath], $command));
    }

    // …
}
```

Reply:
What is this going to solve/improve? Sorry, I'm still not getting it. :upside_down_face:

Reply:
IDK, I just told you what I think @aschempp meant. 😄

Reply:
it just does not require a class with one method in the autoloader just for testing purposes 🙃 
class with one method => function

Reply:
> class with one method => function

That is kind of a point. 👍🏻

Reply:
But testing one path and executing another kind of defeats the purpose of testing, doesn't it?

We could do something like this:
```php

public function __construct(…) {
    $this->processCreationHandler = static function(...$args): Process {
        return new Process(...$args);
    }
}

/**
 * @internal
 */
public function registerProcessCreationHandler(callable $handler) {
    $this->processCreationHandler = $handler;
}

private function executeCommand(array $command, OutputInterface $output): void
{
    $process = $this->processCreationHandler(array_merge([$this->phpPath, $this->consolePath], $command));
    …
}
```

Reply:
Fine with me as well.

Reply:
This now uses a handler like outlined above - I like that the overhead is very small this way.

see 2e06a6ffa7317d8a8b6ba899bd8d004b65e103a7